nixos/modules/services/networking/tetrd.nix

```diff
@@ -0,0 +1,45 @@
+{ config, lib, pkgs, ... }:
+
+{
+  options.services.tetrd.enable = lib.mkEnableOption pkgs.tetrd.meta.description;
+
+  config = lib.mkIf config.services.tetrd.enable {
+    environment.systemPackages = [ pkgs.tetrd ];
+
+    systemd.services.tetrd = {
+      description = pkgs.tetrd.meta.description;
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        mkdir -p /usr/bin
+        ln -sf ${pkgs.nettools}/bin/{route,ifconfig} /usr/bin
+      '';
+
+      serviceConfig = {
+        ExecStart = "${pkgs.tetrd}/opt/Tetrd/bin/tetrd";
+        Restart = "always";
+        RuntimeDirectory = "tetrd";
+        RootDirectory = "/run/tetrd";
+        DynamicUser = true;
+        PermissionsStartOnly = true;
```

Comment:
This is deprecated. You should move your `preStart` script into a file (`pkgs.writeShellScriptBin` or whatever) and then call that script via `ExecStartPre=+/nix/store/xxxxxxx-myscript.sh` if that is what is required. `systemd.tmpfiles.rules` might be a better alternative, though.

Reply:
Sorry, my mistake on `systemd.tmpfiles.rules`! If you have any questions about running `ExecStartPre` scripts as the `root` user please let me know and I can provide an example. Great work!

Reply:
No worries! I appreciate it.
I initially force pushed the wrong version and then re-force-pushed the correct version, so it should be fixed now.
https://github.com/NixOS/nixpkgs/pull/146795/files#diff-35209b17171c21c45b771646355ad42716a9246807f1fae8aad0ca5b3009915fR18

Reply:
Beware that this version still uses the deprecated `PermissionsStartOnly=`.

Reply:
Oh! I wasn't even aware ``PermissionsStartOnly`` was deprecated.
I see what that ``+`` is about now, haha.